invenio_oauthclient/contrib/cern.py

```diff
@@ -247,36 +274,39 @@ def disconnect_handler(remote, *args, **kwargs):
 
 def account_setup(remote, token, resp):
     """Perform additional setup after user have been logged in."""
-    res = get_resource(remote)
+    resource = get_resource(remote)
 
-    groups = fetch_groups(res['Group'])
     with db.session.begin_nested():
-        external_id = res['PersonID'][0]
+        external_id = resource['PersonID'][0]
 
-        # Set CERN person ID in extra_date.
+        # Set CERN person ID in extra_data.
         token.remote_account.extra_data = {
             'external_id': external_id,
-            'groups': groups,
-            # TODO: Add timestamp for refreshing token
         }
+        account_groups(token.remote_account, resource)
         user = token.remote_account.user
 
         # Create user <-> external id link.
         oauth_link_external_id(user, dict(id=external_id, method='cern'))
 
 
-@identity_loaded.connect
-def on_identity_loaded(sender, identity):
+@identity_changed.connect
```

Comment:
@ioannistsanaktsidis please check latest updated version of this PR and code around https://github.com/inveniosoftware/invenio-oauthclient/pull/96/files#diff-b49dd27458efb60f2de74978e6d3eecfR304